core/dbt/node_types.py

```diff
@@ -49,6 +50,10 @@ def documentable(cls) -> List['NodeType']:
             cls.Exposure
         ]
 
+    @classmethod
+    def relational(cls) -> List['NodeType']:
+        return cls.refable() + ([cls.Test] if flags.STORE_FAILURES else [])
```

Comment:
this is.... not ideal... i think. Is there a way that we can push this logic around `flags.STORE_FAILURES` into the caller? Something about changing the behavior of this class method based on a global flag feels pretty fraught

Reply:
Good call. I removed `relational` as a class method and instead implemented this logic as a [`should_store_failures` boolean property of parsed nodes](https://github.com/fishtown-analytics/dbt/pull/3316/files#diff-2865acf9f8c91b43775b17533e2f5ac56f9f6a7bf2fff497e0bc84edaf4af8daR118-R123).

Unfortunately, it's duplicative with logic in the [`should_store_failures()` Jinja macro](https://github.com/fishtown-analytics/dbt/pull/3316/files#diff-4bef0f1be924c18da1624194a38efd20176db97cd0d0bfa6d68f460c7f9e7848R77-R83) used in the `test` materialization, which takes after the `should_full_refresh()` macro. I don't know a good way around this: We need this property accessible in python, for deciding which schemas need to be created as part of the task. I also see merit in making the logic really explicit to users, and allowing them to override it if they'd like. And the places in the codebase where we call Jinja macros from python are... not ideal...